From 56bbb8cc5434e9ed8ecf7392cc4a3a330cedd92e Mon Sep 17 00:00:00 2001
From: Karim Pinter <karim.pinter@qt.io>
Date: Tue, 30 Sep 2025 13:57:27 +0300
Subject: [PATCH 426/553] Fix implicit integer conversion loses build error on
 Intel VxWorks

Building Qt on VxWorks Intel platform results in implicit conversion
loses integer precision build error. According to [1] the cast should
be ok.

[1] https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_extract_epi16&ig_expand=2927

Amends c817b33b45f2886c89d8a768c61ffffaa46f67a4.

Fixes: QTBUG-140672
Pick-to: 6.8 6.5
Change-Id: Ic99dedac0cfc97ac84a9d0c9df24259bdab23e80
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
Reviewed-by: Marc Mutz <marc.mutz@qt.io>
(cherry picked from commit c0303a9630fd7876280b840dea6fc9ead88d86da)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/global/qfloat16.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/global/qfloat16.h b/src/corelib/global/qfloat16.h
index 09b52026b57..b8d438d1ccf 100644
--- a/src/corelib/global/qfloat16.h
+++ b/src/corelib/global/qfloat16.h
@@ -359,7 +359,7 @@ inline qfloat16::qfloat16(float f) noexcept
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__F16C__)
     __m128 packsingle = _mm_set_ss(f);
     __m128i packhalf = _mm_cvtps_ph(packsingle, 0);
-    b16 = _mm_extract_epi16(packhalf, 0);
+    b16 = quint16(_mm_extract_epi16(packhalf, 0));
 #elif defined (__ARM_FP16_FORMAT_IEEE)
     __fp16 f16 = __fp16(f);
     memcpy(&b16, &f16, sizeof(quint16));
-- 
2.51.2

